internal/services/machinelearning/machine_learning_compute_cluster_resource.go

```diff
@@ -355,6 +357,84 @@ func resourceComputeClusterRead(d *pluginsdk.ResourceData, meta interface{}) err
+	mlWorkspacesClient := meta.(*clients.Client).MachineLearning.Workspaces
+	client := meta.(*clients.Client).MachineLearning.MachineLearningComputes
+	ctx, cancel := timeouts.ForCreate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	workspaceID, err := workspaces.ParseWorkspaceID(d.Get("machine_learning_workspace_id").(string))
+	if err != nil {
+		return err
+	}
+
+	id, err := machinelearningcomputes.ParseComputeID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	workspace, err := mlWorkspacesClient.Get(ctx, *workspaceID)
+	if err != nil {
+		return fmt.Errorf("retrieving %s: %+v", workspaceID, err)
+	}
+
+	workspaceModel := workspace.Model
+	if workspaceModel == nil {
+		return fmt.Errorf("retrieving %s: `model` was nil", workspaceID)
+	}
+
+	identity, err := expandIdentity(d.Get("identity").([]interface{}))
+	if err != nil {
+		return fmt.Errorf("expanding `identity`: %+v", err)
+	}
+
+	vmPriority := machinelearningcomputes.VMPriority(d.Get("vm_priority").(string))
+	computeClusterAmlComputeProperties := machinelearningcomputes.AmlComputeProperties{
+		VMSize:                 utils.String(d.Get("vm_size").(string)),
+		VMPriority:             &vmPriority,
+		ScaleSettings:          expandScaleSettings(d.Get("scale_settings").([]interface{})),
+		UserAccountCredentials: expandUserAccountCredentials(d.Get("ssh").([]interface{})),
+		EnableNodePublicIP:     pointer.To(d.Get("node_public_ip_enabled").(bool)),
+	}
+
+	computeClusterAmlComputeProperties.RemoteLoginPortPublicAccess = pointer.To(machinelearningcomputes.RemoteLoginPortPublicAccessDisabled)
+	if d.Get("ssh_public_access_enabled").(bool) {
+		computeClusterAmlComputeProperties.RemoteLoginPortPublicAccess = pointer.To(machinelearningcomputes.RemoteLoginPortPublicAccessEnabled)
+	}
+
+	if subnetId, ok := d.GetOk("subnet_resource_id"); ok && subnetId.(string) != "" {
+		computeClusterAmlComputeProperties.Subnet = &machinelearningcomputes.ResourceId{Id: subnetId.(string)}
+	}
+
+	// NOTE: The 'AmlCompute' 'ComputeLocation' field should always point
+	// to configuration files 'location' field...
+	computeClusterProperties := machinelearningcomputes.AmlCompute{
+		Properties:       &computeClusterAmlComputeProperties,
+		ComputeLocation:  utils.String(d.Get("location").(string)),
+		Description:      utils.String(d.Get("description").(string)),
+		DisableLocalAuth: utils.Bool(!d.Get("local_auth_enabled").(bool)),
+	}
+
+	// NOTE: The 'ComputeResource' 'Location' field should always point
+	// to the workspace's 'location'...
+	computeClusterParameters := machinelearningcomputes.ComputeResource{
+		Properties: computeClusterProperties,
+		Identity:   identity,
+		Location:   workspaceModel.Location,
+		Tags:       tags.Expand(d.Get("tags").(map[string]interface{})),
+		Sku: &machinelearningcomputes.Sku{
+			Name: workspaceModel.Sku.Name,
+			Tier: pointer.To(machinelearningcomputes.SkuTier(*workspaceModel.Sku.Tier)),
+		},
+	}
```

Comment:
We should still be able to retrieve the existing Compute Cluster and patch the SKU from the workspace into the model instead of having to set everything from the config like in the create?

Reply:
or we can try update without SKU?  if this works we don't need to get the workspace in update

Reply:
Sure, try it

Reply:
Tried, can update without SKU. Also the API doc the SKU is the SKU for the workspace(https://learn.microsoft.com/en-us/rest/api/azureml/compute/create-or-update?view=rest-azureml-2024-04-01&tabs=HTTP#request-body).  So we can ignore it or get it from the workspace, do we need to remove it in update?

Reply:
If you can update successfully without sending anything for the SKU (and it doesn't change the SKU) then it's fine to omit getting it from the workspace.